.github/workflows/ci.yaml

```diff
@@ -20,6 +20,106 @@ concurrency:
+      test_full_suite: ${{ steps.info.outputs.test_full_suite }}
+      core: ${{ steps.core.outputs.changes }}
+      integrations: ${{ steps.integrations.outputs.changes }}
+      integrations_glob: ${{ steps.info.outputs.integrations_glob }}
+      tests: ${{ steps.info.outputs.tests }}
+      tests_glob: ${{ steps.info.outputs.tests_glob }}
+      test_groups: ${{ steps.info.outputs.test_groups }}
+      test_group_count: ${{ steps.info.outputs.test_group_count }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code from GitHub
+        uses: actions/checkout@v2.4.0
+      - name: Filter for core changes
+        uses: dorny/paths-filter@v2.10.2
+        id: core
+        with:
+          filters: .core_files.yaml
+      - name: Create a list of integrations to filter for changes
+        run: |
+          integrations=$(ls -Ad ./homeassistant/components/[!_]*  | xargs -n 1 basename)
+          touch .integration_paths.yaml
+          for integration in $integrations; do
+            echo "${integration}: [homeassistant/components/${integration}/*, tests/components/${integration}/*]" \
+              >> .integration_paths.yaml;
+          done
+          echo "Result:"
+          cat .integration_paths.yaml
+      - name: Filter for integration changes
+        uses: dorny/paths-filter@v2.10.2
+        id: integrations
+        with:
+          filters: .integration_paths.yaml
+      - name: Collect additional information
+        id: info
+        run: |
+          # Defaults
+          integrations_glob=""
+          test_full_suite="true"
+          test_groups="[1, 2, 3, 4, 5, 6]"
+          test_group_count=6
+          tests="[]"
+          tests_glob=""
+
+          if [[ "${{ steps.integrations.outputs.changes }}" != "[]" ]];
+          then
+            # Create a file glob for the integrations
+            integrations_glob=$(echo '${{ steps.integrations.outputs.changes }}' | jq -cSr '. | join(",")')
+            [[ "${integrations_glob}" == *","* ]] && integrations_glob="{${integrations_glob}}"
+
+            # Create list of testable integrations
+            possible_integrations=$(echo '${{ steps.integrations.outputs.changes }}' | jq -cSr '.[]')
+            tests=$(
+              for integration in ${possible_integrations};
+              do
+                if [[ -d "tests/components/${integration}" ]]; then
+                  echo -n "\"${integration}\",";
+                fi;
+              done
+            )
+
+            [[ ! -z "${tests}" ]] && tests="${tests::-1}"
+            tests="[${tests}]"
+            test_groups="${tests}"
+            # Test group count should be 1, we don't split partial tests
+            test_group_count=1
+
+            # Create a file glob for the integrations tests
+            tests_glob=$(echo "${tests}" | jq -cSr '. | join(",")')
+            [[ "${tests_glob}" == *","* ]] && tests_glob="{${tests_glob}}"
+
+            test_full_suite="false"
+          fi
```

Comment:
In the future (not for this PR) would be nicer if this was generated by a python script so this would be easier to test.  Reading this gives me a bit of a headache.

Reply:
For me its actually the opposite 😄